qiskit/primitives/base_sampler.py

```diff
@@ -0,0 +1,155 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+r"""
+=======
+Sampler
+=======
+
+Sampler class estimates quasi-probabilities of bitstrings from quantum circuits.
+
+The input consists of following elements.
+
+* quantum circuits (:math:`\psi_i(\theta)`): list of (parametrized) quantum circuits.
+  (a list of :class:`~qiskit.circuit.QuantumCircuit`))
+
+* parameters: a list of parameters of the quantum circuits.
+  (:class:`~qiskit.circuit.parametertable.ParameterView` or
+  a list of :class:`~qiskit.circuit.Parameter`).
+
+* parameter values (:math:`\theta_k`): 1-dimensional or 2-dimensional array of values
+  to be bound to the parameters of the quantum circuits.
+  (list of float or list of list of float)
+
+The output is the quasi-probabilities of bitstrings.
+
+The sampler object is expected to be initialized with "with" statement
+and the objects are called with parameter values and run options
+(e.g., ``shots`` or number of shots).
+
+Here is an example of how sampler is used.
+
```

Comment:
i feel like the base file does not need an example. each of the implementations (quantum_info/aer/ibm/..) can have their own example as they may be slightly different (quantum_info/aer have exact estimators, ibm has error mitigation, etc.)

The example you have here uses backend which was removed from base.

Reply:
I removed the ref to backend